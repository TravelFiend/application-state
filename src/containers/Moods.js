import React, { useReducer } from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import incrementorsReducer from '../reducers/incrementorsReducer';
import getFace from '../selectors/emojiSelectors';
import { incrementCoffees, incrementSnacks, incrementNaps, incrementStudies } from '../actions/incrementors';

export default function Moods() {
  const [state, dispatch] = useReducer(incrementorsReducer, { coffees: 0, snacks: 0, naps: 0, studies: 0 });

  const actions = [incrementCoffees, incrementSnacks, incrementNaps, incrementStudies];
  const face = getFace(state);

  const controlActions = actions.map(action => action);

  return (
    <>
      <Controls actions={controlActions} handleSelection={type => dispatch(type())}/>
      <Face emoji={face} />
    </>
  );
}
