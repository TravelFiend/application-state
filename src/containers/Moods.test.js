import React from 'react';
import { shallow } from 'enzyme';
import Moods from './Moods';
// import Moods from '../selectors/emojiSelectors';

describe('container', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Moods />);
    expect(wrapper).toMatchSnapshot();
  });
});
