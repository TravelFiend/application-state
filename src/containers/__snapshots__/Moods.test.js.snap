// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container matches a snapshot 1`] = `"ðŸ˜€"`;
